independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -300,10 +300,8 @@ private ClassLoaderState getState() {
                             if (transformedClasses.getResource(i) != null) {
                                 elementMap.put(i, Collections.singletonList(transformedClasses));
                             } else {
-                                List<ClassPathElement> list = elementMap.get(i);
-                                if (list == null) {
-                                    elementMap.put(i, list = new ArrayList<>(2)); //default initial capacity of 10 is way too large
-                                }
+                                //default initial capacity of 10 is way too large
+                                List<ClassPathElement> list = elementMap.computeIfAbsent(i, key -> new ArrayList<>(2));
```

Comment:
No lambdas in runtime code.

Reply:
right :) thanks

Reply:
This is not a capturing lambda so perf is on par with no lambda.
And computeIfAbsent is already used in other runtime part.
But I'll revert the change if you prefere, it is not perf related.

Reply:
As you say capturing lambdas are the "top enemy" as they allocate at runtime, but we're mindful for other lambdas as well as they require a new class to be defined and loaded - with all security checks and verifications of the case.
It's just not worth it in code like this.

There's of course use of lambdas at runtime as many libraries and user code are likely to use them, but we'd still prefer to be mindful about them in crucial infastructure code. It's of course a matter of judgment and tradoffs.